test/test-fs-event.c

```diff
@@ -184,38 +144,76 @@ static const char* fs_event_get_filename_in_subdir(int i) {
 static void fs_event_create_files(uv_timer_t* handle) {
   int i;
 
-  /* Already created all files */
-  if (fs_event_created == fs_event_file_count) {
-    uv_close((uv_handle_t*) &timer, close_cb);
-    return;
-  }
-
   /* Create all files */
   for (i = 0; i < 16; i++, fs_event_created++)
     create_file(handle->loop, fs_event_get_filename(i));
 
-  /* And unlink them */
-  ASSERT(0 == uv_timer_start(&timer, fs_event_unlink_files, 50, 0));
+  /* Close the handle */
+  uv_close((uv_handle_t*) &timer, close_cb);
+  return;
 }
 
 static void fs_event_create_files_in_subdir(uv_timer_t* handle) {
   int i;
 
-  /* Already created all files */
-  if (fs_event_created == fs_event_file_count) {
-    uv_close((uv_handle_t*) &timer, close_cb);
-    return;
-  }
-
   /* Create all files */
   for (i = 0; i < 16; i++, fs_event_created++)
     create_file(handle->loop, fs_event_get_filename_in_subdir(i));
 
-  /* And unlink them */
-  ASSERT(0 == uv_timer_start(&timer, fs_event_unlink_files_in_subdir, 50, 0));
+  /* Close the handle */
+  uv_close((uv_handle_t*) &timer, close_cb);
+  return;
 }
 
-void fs_event_unlink_files_in_subdir(uv_timer_t* handle) {
+static void fs_event_cb_dir_multi_file(uv_fs_event_t* handle,
+                                       const char* filename,
+                                       int events,
+                                       int status) {
+  fs_event_cb_called++;
+  ASSERT(handle == &fs_event);
+  ASSERT(status == 0);
+  ASSERT(events == UV_RENAME);
+  ASSERT(filename == NULL ||
+         strncmp(filename, file_prefix, sizeof(file_prefix) - 1) == 0);
+
+  fprintf(stderr, "fs_event_cb_dir_multi_file called: %i\n", fs_event_cb_called);
+  fprintf(stderr, "  created: %i\n", fs_event_created);
+  fprintf(stderr, "  removed: %i\n", fs_event_removed);
+
+  if (fs_event_cb_called == fs_event_created) {
+    /* Once we've processed all create events, delete all files */
+    fs_event_unlink_files(handle);
+  } else if (fs_event_cb_called == fs_event_created + fs_event_removed) {
+    /* Once we've processed all create and delete events, stop watching */
+    ASSERT(0 == uv_fs_event_stop(handle));
```

Comment:
Makes sense.  I didn't write the original test so I didn't want to change it but I will remove it if you think it's unnecessary.